### DISCUSSION TITLE
setResponseStatus not working from 404 to 200

### DISCUSSION THREAD

**Discussion by Rekwian** [nuxt/nuxt#24917] [General]
Hello

I'm trying to set a proxy on nuxt error, the use case : when nuxt returns 404 error => proxy to wordpress.
With a defineEventHandler, it works, but I can't change status code to 200 with setResponseStatus.
If I try to change status with 202, it works, but code 200 still fails and return 404.

```javascript
export default defineEventHandler(async (event) => {
  const url = getRequestURL(event);

  if (url.pathname.match('^/__nuxt_error')) {
    const proxyUrl = new URL(`http://dev.wp.my-app.com/${url.searchParams.get('url')}`)
    const proxy = await proxyRequest(event, proxyUrl)

    console.log({ proxy: [proxy.statusCode, proxy.statusMessage] }) // { proxy: [ 200, 'OK' ] }

    setResponseStatus(event, proxy.statusCode, proxy.statusMessage)

    console.log({ eventNode : event.node.res.statusCode }) // { eventNode: 200 }

    return proxy
  }
});
```

An issue was created in unjs/h3 GitHub, but unanswered. https://github.com/unjs/h3/issues/541
Maybe the statusCode is overrided by nuxt, but I don't know where.

Any idea to help me ?

**Comment by TheAlexLichter** [comment]
Answered in the linked issue. Did you also try it via CURL? Can you check via `curl -I -X GET http://localhost:3000/my-url-here`?

**Reply by Rekwian** [reply]
![image](https://github.com/nuxt/nuxt/assets/31690329/235bd37c-9ee4-44b8-aea8-5fdbe615afb5)

**Reply by Rekwian** [reply]
However, curl does not trigger the defineEventHandler for /__nuxt_error. this is a problem for me, this proxy must be viewed from the bots.
Maybe nuxt/nitro/h3 has another solution to set a proxy when 404 is thrown and make it visible to bots ?

**Reply by TheAlexLichter** [reply]
I can reproduce this with the app provided. Might be a better way to proxy "on error" without overriding but I have to check that later on.
